### DISCUSSION TITLE
overwrite existing resources

### DISCUSSION THREAD

**Discussion by thulasidassrinivasan** [vmware-tanzu/velero#3781] [Community support Q&A] [answered]
Hi, 

Is there any way velero can overwrite existing resources when running velero restore ?  I would like to achieve it. 

for e.g., I have service A that talks to pod A. I back up and restore this from cluster A to Cluster B and after some testing, I see the svc A in cluster A is updated from deployments. So I take a backup again from cluster A but restoring that in cluster B is not working as cluster B already have the svc in same name so velero doesn't overwrite that.

Version: v1.6.0
velero-plugin-for-microsoft-azure:v1.2.0

**Comment by thulasidassrinivasan** [comment]
@zubron or @jenting - Any advise here please ?

**Reply by jenting** [reply]
According to the Velero design, it doesn't allow overwriting existing resources.

**Reply by thulasidassrinivasan** [reply]
@jenting thanks.

**Comment by jenting** [comment, marked as answer]
According to the Velero design, it doesn't allow overwriting existing resources.
Related issues
- https://github.com/vmware-tanzu/velero/issues/74
- https://github.com/vmware-tanzu/velero/issues/1845#issuecomment-528400792

**Reply by thulasidassrinivasan** [reply]
Since this was discussed few years back (2017) I thought this will be incorporated in some release of velero.  I am not sure why this was not implemented, but I think this is a great feature to have something like a flag `--overwrite-existing-resource` and the person doing the restore can decide to pass boolean `yes or no` when doing restore will be helpful.

**Reply by jenting** [reply]
Totally agree.
